.github/workflows/release.yaml

```diff
@@ -15,16 +15,24 @@
 name: release-guac-image
 
 on:
-  release:
-    types: [published]
+  push:
+    tags:
+      - 'v*'
```

Comment:
Do we want to create releases manually or have the releases created by pushing tags?

Reply:
It looks like a tag is a pre-req of creating a release. I changed the trigger of the image build workflow to react to tag push. So we could potentially test the image prior to cutting a release or build beta images that is not intended for a release.

We could add logic to auto-generate the release if we want. Right now we can still cut the release manually and have the image tag matches it.

Reply:
Let's add a doc (release.md / developer.md ?) with our intended workflow:

1. Push tag
2. verify images
3. Create release in GitHub

... etc.

Reply:
Yup, makes sense and we should make a decision on how we want to do 3.

Reply:
Added #791